bundle/src/main/java/com/adobe/acs/commons/packaging/impl/AssetPackagerServletImpl.java

```diff
@@ -0,0 +1,281 @@
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.wrappers.ValueMapDecorator;
+
+/**
+ * ACS AEM Commons - Asset Packager Servlet
+ * Servlet end-point used to create packages of pages and the referenced assets.
+ */
+@SuppressWarnings("serial")
+@SlingServlet(
+        methods = {"POST"},
+        resourceTypes = {"acs-commons/components/utilities/packager/asset-packager"},
+        selectors = {"package"},
+        extensions = {"json"}
+)
+public class AssetPackagerServletImpl extends AbstractPackagerServlet {
+
+    /* Property names */
+    private static final String PN_PAGE_PATH = "pagePath";
+    private static final String PN_EXCLUDE_PAGES = "excludePages";
+    private static final String PN_ASSET_PREFIX = "assetPrefix";
+    private static final String PN_PAGE_EXCLUSIONS = "pageExclusions";
+    private static final String PN_ASSET_EXCLUSIONS = "assetExclusions";
+
+    /* Default Properties */
+    private static final String DEFAULT_PACKAGE_NAME = "assets";
+    private static final String DEFAULT_PACKAGE_GROUP_NAME = "Assets";
+    private static final String DEFAULT_PACKAGE_DESCRIPTION = "Asset Package initially defined by a ACS AEM Commons - "
+            + "Asset Packager configuration.";
+    private static final String THUMBNAIL_RESOURCE_PATH =
+            "/apps/acs-commons/components/utilities/packager/asset-packager/definition/package-thumbnail.png";
+
+    @Reference
+    private PackageHelper packageHelper;
+
+    private String customPrefix;
+    private List<Pattern> pageExclusionPatterns;
+    private List<Pattern> assetExclusionPatterns;
+    private final List<String> excludedPages = new ArrayList<>();
+
+    @Override
+    public final void doPost(final SlingHttpServletRequest request,
+                             final SlingHttpServletResponse response) throws IOException {
+
+        final ResourceResolver resourceResolver = request.getResourceResolver();
+        final boolean preview = Boolean.parseBoolean(request.getParameter("preview"));
+
+        log.debug("Preview mode: {}", preview);
+
+        final ValueMap properties = this.getProperties(request);
+
+        this.customPrefix = properties.get(PN_ASSET_PREFIX, String.class);
+        this.pageExclusionPatterns = generatePatterns(properties.get(PN_PAGE_EXCLUSIONS, new String[]{}));
+        this.assetExclusionPatterns = generatePatterns(properties.get(PN_ASSET_EXCLUSIONS, new String[]{}));
+
+        try {
+            final String pagePath = properties.get(PN_PAGE_PATH, String.class);
+            final List<PathFilterSet> paths;
+            if (StringUtils.isBlank(pagePath)) {
+                paths = Collections.emptyList();
+            } else {
+                paths = this.findAssetPaths(resourceResolver, pagePath);
+            }
+            if (!properties.get(PN_EXCLUDE_PAGES, false) && paths.size() > 0) {
```

Comment:
For code consistency and readability, would it make sense to put lines 111-119 into a `findPagePaths` function similar to `findAssetPaths`?  seems odd that we delegate for one and not the other.

Reply:
After reading the code I have a better understanding of what's happening here...this is more `addPageExcludes`

Reply:
It's more of a check to see if the user wants the page path included in the content package, and if they do then we add the page + any exclusions they added. I'll separate it out into another method

Reply:
Addressed this!